packages/jest-jasmine2/src/expectation_result_factory.js

```diff
@@ -36,8 +36,14 @@ function stackFormatter(options, initError, errorMessage) {
     return '';
   }
 
-  if (options.error && options.error.stack) {
-    return options.error.stack;
+  if (options.error) {
+    if (options.error.stack) {
+      return options.error.stack;
+    }
+
+    if (options.error === errorMessage) {
+      return errorMessage;
```

Comment:
the assumption is here that this means that we already have a stack trace, and don't need to add our extra one